shell/browser/ui/file_dialog_gtk.cc

```diff
@@ -272,8 +339,25 @@ bool ShowOpenDialogSync(const DialogSettings& settings,
   FileChooserDialog open_dialog(action, settings);
   open_dialog.SetupProperties(settings.properties);
 
-  gtk_widget_show_all(open_dialog.dialog());
-  int response = gtk_dialog_run(GTK_DIALOG(open_dialog.dialog()));
+  if (GTK_IS_WIDGET(open_dialog.dialog())) {
+    gtk_widget_show_all(GTK_WIDGET(open_dialog.dialog()));
+  } else {
+    void (*dl_gtk_native_dialog_show)(void*) = nullptr;
+    g_module_symbol(open_dialog.gtk_module(), "gtk_native_dialog_show",
+                    reinterpret_cast<void**>(&dl_gtk_native_dialog_show));
+    dl_gtk_native_dialog_show(static_cast<void*>(open_dialog.dialog()));
+  }
```

Comment:
We have this logic twice now. Maybe a new helper `ShowFileDialog(void*)` similar to the `SetGtkTransientForAura()` helper